activerecord/lib/active_record/associations/collection_proxy.rb

```diff
@@ -31,6 +31,9 @@ class CollectionProxy < Relation
       def initialize(klass, association) #:nodoc:
         @association = association
         super klass, klass.arel_table, klass.predicate_builder
+
+        extensions = association.extensions
+        extend(*extensions) if extensions.any?
```

Comment:
I extracted better-performing `association.extensions` as @matthewd's suggestion.
It is 2x slower than caching extensions, but seems acceptable. WDYT?

Caching on:

```
CollectionProxies (1000000):   0.740000   0.000000   0.740000 (  0.737209)
Associations (1000000):   0.110000   0.000000   0.110000 (  0.110163)
```

Caching off:

```
CollectionProxies (1000000):   1.620000   0.000000   1.620000 (  1.629061)
Associations (1000000):   0.110000   0.000000   0.110000 (  0.112852)
```

Benchmark script: https://gist.github.com/kevingriffin/29d7e4774d210a41116c063dfd430de0 in #28656.